src/wazuh_db/helpers/wdb_global_helpers.h

```diff
@@ -108,21 +108,31 @@ int wdb_update_agent_data(agent_info_data *agent_data, int *sock);
  * @param[in] id Id of the agent for whom the keepalive must be updated.
  * @param[in] connection_status String with the connection status to be set.
  * @param[in] sync_status String with the cluster synchronization status to be set.
+ * @param [in] disconnected_time The date an agent gets disconnected from the manager.
  * @param[in] sock The Wazuh DB socket connection. If NULL, a new connection will be created and closed locally.
  * @return OS_SUCCESS on success or OS_INVALID on failure.
  */
-int wdb_update_agent_keepalive(int id, const char *connection_status, const char *sync_status, int *sock);
+int wdb_update_agent_keepalive(int id,
+                               const char *connection_status,
+                               const char *sync_status,
+                               time_t disconnected_time,
```

Comment:
I wouldn't add this parameter here. When we update the keepalive, it means that the agent will be in `PENDING` or `ACTIVE` state. In both states, the disconnected time should be set to `0`. So, I suggest you to remove this and directly set the disconnected time to `0` in the SQL query here:

https://github.com/wazuh/wazuh/pull/10102/files#diff-dc655e05833340d75b128c0db4c09a2f42a17fa0c11240fce745cef50752669dR139

Reply:
As we agreed with the team, we will remove this parameter from the helper method. Then, after parsing the command in Wazuh DB and identifying that the status to be set is `ACTIVE` or `PENDING`, it will bind a `0` (zero) value to the disconnected time in the SQL statement.

Reply:
Changed according to what we discussed